tasks/tasks.go

```diff
@@ -0,0 +1,93 @@
+package tasks
+
+import (
+	"errors"
+	"math/rand"
+	"sync"
+	"time"
+)
+
+// Status is the status of the task.
+type Status int
+
+// TaskExecutor executes a series of task functions.
+type TaskExecutor interface {
+	QueueTask(task func() error) string
+	GetTaskStatus(id string) (Status, error)
```

Comment:
you can make string id's have a fixed length which is better if you need to store it somewhere.